modules/system/assets/ui/js/select.js

```diff
@@ -86,7 +86,23 @@
 
                         return $request
                     },
-
+                    processResults: function (data, params) {
+                        var results = JSON.parse(data.result);
+                        var options = [];
+
+                        for (var i in results) {
+                            if (results.hasOwnProperty(i)) {
+                                options.push({
+                                    id: i,
+                                    text: results[i],
+                                });
+                            };
+                        };
+
+                        return {
+                            results: options,
```

Comment:
here, I think key should be "result"

Reply:
Ah, well, I'm basing this off of the [select2 docs for transforming data](https://select2.org/data-sources/ajax#transforming-response-data). 

I just tested with this code:
```
processResults: function (data,params) {
                        var results = data.result;
                        var options = [];

                        for (var i in results) {
                            if (results.hasOwnProperty(i)) {
                                options.push({
                                    id: i,
                                    text: results[i],
                                });
                            };
                        };                       
                        return {
                            results: options
                        };
```

And it is working. The only thing is that the items are not actually filtered based on your input. You just get a list of all options.

Reply:
ok, my bad.

Reply:
No worries. Any thoughts on why the items are not filtered based on input? Is this something that has to happen in the backend?